### ISSUE TITLE
Personal preference on why JAX

### ISSUE THREAD

**Issue by alirezakazemipour** [danijar/ninjax#3]
Hi Danijar,
I hope you're doing well!
As I have learned recently, the research community is heading more toward using JAX as the preferred framewrok to do Machine Learning research especially, when Neural Computing is also involved.
I skimmed through your past research works and I found that although, you used to use TensorFlow or PyTorch in papers like Dreamer or Dreamer V2 but now, you have come up with having a general API (ninjax) to probably incorporate JAX into your recent projects.
My question is, why have you reached to a point to use more of JAX? What do features of it seem intriguing to you?

Personally, I'd like to learn it as well but, I'm having difficulties at these first steps at the beginning and more importantly, I feel all the time that JAX doesn't add anything new to what TensorFlow/PyTorch have already managed to offer.
For instance, the `jax.jit()` transform is completely a reminiscent of `tensorflow.function()` to me thus, vague to leave something well-developed like TensorFlow behind to use JAX.

I'd be really grateful to have your idea. Absolutely, it will help me to pay attention to what might have been unnoticed to me.

**Comment by EelcoHoogendoorn** [comment]
Just came across it; and as someone who has used everything from theano to tf to torch to now jax, I can answer from my perspective.

What I like about jax is that it providers a clean seperation between the 'language' and the 'ml framework'. Both torch and tf lean very much in the direction of wanting to be a useful ML framework first and foremost, and it allows its low level framework decision such as 'the first axis is the batch axis' or 'the last axis is the feature axis' to permeate throughout its design.

JAX is first and foremost an array computing language. I personally use it for all kinds of non-ml workloads; and it has never been the case that its intended ML use case gets in the way of my non-ml use cases, overtly or subtly. 

It is also this seperation and generality that allows for the wonders of arbitrarily composable vmap, jit and grad operations. torch may have something resembling jit; but there is no chance it will ever be anything but a giant ball of corner cases.

That being said I wouldnt recommend using JAX as a quick win; and I did hold off on switching my team to JAX in the specific circumstances I was working on a year ago despite my personal enthusiasm. As a long term investment in my own computing skills, I cant recommend it enough though.

**Comment by alirezakazemipour** [comment]
Hi @EelcoHoogendoorn,
Thank you so much for your invaluable share of thoughts and experiences!
Now, I see it in a much clearer way. 
Awesome!
